tutorials/intellijDebugger.md

```diff
@@ -6,7 +6,89 @@ pageNav: 3
 
 # {{ title }}
 
+Before diving into tracing code with IntelliJ IDEA’s debugger, it’s essential to understand key debugger features and their purpose.
+
 This video (from LaunchCode) gives a pretty good explanation of how to use the IntelliJ IDEA debugger.
 
 @[youtube](1bCgzjatcr4)
 
+## Starting A Debugging Session
+
+To utilise the debugger, we need to make sure that the code is running in debugging mode.
+
+### Run Debugger Mode
+
+<box type="info" seamless>
+
+**Video reference:** [3.53](https://youtu.be/1bCgzjatcr4?si=fIEvUVhXHwKQeOlc&t=233): Run your program in debugging mode to use debugging features.
+</box>
+
+Go to `Run` > `Debug 'Main'` or click on the bug icon in the top-right corner of the IntelliJ IDEA window.
+
+### Setting Breakpoints
+
+<box type="info" seamless>
+
+**Video reference:** [3.09](https://youtu.be/1bCgzjatcr4?si=KkJw2Cri-ERmqgAJ&t=189): Set a breakpoint by clicking on the line number in the left gutter like this!
+</box>
+
+Purpose: Breakpoints pause program execution at specific lines of code.
+
+Usage: Place breakpoints at critical points where you want to examine the program state, such as method entry points or before complex operations.
```

Comment:
Just some thoughts: 
I was thinking that perhaps the flow might be better if we had a quick introduction to the mode before going into the video section!
Maybe we could also use bullet points to keep it short and concise for each section?
```
Breakpoints allow you to pause program execution at specific lines of code, making it easier to examine the state of your application at those points.

<box type="info" seamless>
**Video reference:** [3.09](https://youtu.be/1bCgzjatcr4?si=KkJw2Cri-ERmqgAJ&t=189): Set a breakpoint by clicking on the line number in the left gutter like this!
</box>

To set a breakpoint in IntelliJ IDEA:
- Click on the line number in the left gutter.

Breakpoints are useful for:
- Pausing execution at method entry points
- Examining the program state before complex operations
```

Reply:
Thanks for the suggestion!
Modified the sections with some bullet points and examples!